x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/fielddata/ScriptBooleanDocValues.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.fielddata;
+
+import org.elasticsearch.index.fielddata.AbstractSortedNumericDocValues;
+import org.elasticsearch.xpack.runtimefields.BooleanScriptFieldScript;
+
+import java.io.IOException;
+
+public final class ScriptBooleanDocValues extends AbstractSortedNumericDocValues {
+    private final BooleanScriptFieldScript script;
+    private int cursor;
+
+    ScriptBooleanDocValues(BooleanScriptFieldScript script) {
+        this.script = script;
+    }
+
+    @Override
+    public boolean advanceExact(int docId) {
+        script.runForDoc(docId);
+        cursor = 0;
+        return script.trues() > 0 || script.falses() > 0;
+    }
+
+    @Override
+    public long nextValue() throws IOException {
+        // Emit all false values before all true values
+        return cursor++ < script.falses() ? 0 : 1;
```

Comment:
I must say that multiple values for booleans look really weird, especially trying to summarize multiple values into one, and re-sorting them. Is this behaviour documented anywhere?

Reply:
I'm not sure it is, though it is what happens when you use doc values for out of the box booleans.